### Задание 1
Известно, что генеральная совокупность распределена нормально
со средним квадратическим отклонением, равным 16.
Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95,
если выборочная средняя M = 80, а объем выборки n = 256.

In [78]:
from math import sqrt
sigma = 16 # сигма известна, значит используем Z
a = 0.05
p = 1 - a
M = 80
n = 256
# a/2 = 0.025 Находим Z по таблице для значения 1 - a = 0.975
z = 1.96


In [79]:
X1 = M + (z*sigma/sqrt(n))
X1

81.96

In [80]:
X2 = M - (z*sigma/sqrt(n))
X2

78.04

### Ответ [78.04; 81.96]

## Задание 2
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью,
получены опытные данные:
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей,
оценить истинное значение величины X при помощи доверительного интервала, покрывающего это
значение с доверительной вероятностью 0,95.

In [81]:
import pandas as pd
data = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
df = pd.DataFrame({'data':data})
a = 0.05
n = 10
p = 0.95
E = df['data'].mean()
E

6.589999999999999

In [82]:
# Находим несмещенную дисперсию
D = df['data'].var(ddof=1)
D

0.2032222222222223

In [84]:
# среднее квадратичное несмещенной дисперсии
s = sqrt(D)
s

0.4508017549014448

Поскольку сигма неизвестна, используем параметр t в формуле $$\overline{X} \pm t_{таб}\frac{s}{\sqrt{n}}$$

Находим t по таблице с параметрами (n-1;α/2) = (9;0.025) = 2.262

In [85]:
X1 = E + 2.262 * (s /sqrt(n))
X1

6.912461744087573

In [86]:
X2 = E - 2.262 * (s /sqrt(n))
X2

6.267538255912425

### Ответ: Истинное значение X лежит в доверительном диапазоне  [6.2675; 6.9125]

### Задание 3
Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.


In [87]:
H0 = 17
H1 = 17.5
a = 0.05
n = 100
D = 4
sigma = sqrt(D)

In [90]:
# Сигма известна, выбираем Z по таблице (1,6; 0.045) = 1.645
Zt = 1.645
Zn = (H1 - H0) / (sigma / sqrt(n))
Zn

2.5

### Ответ: Zn > Zt, соответственно гипотеза неверна

### Задача 4
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [91]:
veight = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
df = pd.DataFrame({'veight':veight})
a = 0.01
n = 10
p = 0.99
X = df['veight'].mean()
X

198.5

In [92]:
# Находим несмещенную дисперсию
D = df['veight'].var(ddof=1)
D

19.833333333333332

In [93]:
# среднее квадратичное
s = df['veight'].std(ddof=1)
s

4.453463071962462

In [94]:
# Гипотеза H0: а = 200, Н1: а ≠ 200
H0 = 200

Сигма не задана - используем параметр t. Находим t по таблице с параметрами (n-1;α/2) = (9;0.005) = 3.250

In [95]:
t = 3.250

In [96]:
Tn = (X - H0) / (s / sqrt(n))
abs(Tn)

1.0651074037450896

### Ответ: | Tn | < t, соответственно гипотеза H0 верна